File: mobile-app/app/src/main/java/com/example/fr/data/models/Shelter.kt
package com.example.fr.data.models

import kotlinx.serialization.Serializable

@Serializable
data class Shelter(
    val id: Int,
    val name: String,
    val description: String? = null,
    val address: String? = null,
    val phone: String? = null,
    val latitude: Double,
    val longitude: Double,
    val capacity: Int? = null,
    val current_occupancy: Int = 0,
    val available_space: Int? = null,
    val status: String = "open",
    val is_verified: Boolean = false,
    val created_at: String? = null,
    val updated_at: String? = null,
    // Legacy fields for backward compatibility
    val incident_type: String = "Shelter",
    val report_time: String? = null,
    val user_name: String? = null,
    val verification_count: Int = 0
)

@Serializable
data class ShelterRequest(
    val name: String,
    val description: String,
    val address: String? = null,
    val phone: String? = null,
    val latitude: Double,
    val longitude: Double,
    val capacity: Int? = null
)

@Serializable
data class ShelterResponse(
    val message: String,
    val shelter: Shelter
)

File: backend/app/Models/Shelter.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Shelter extends Model
{
    protected $fillable = [
        'name',
        'description',
        'address',
        'phone',
        'latitude',
        'longitude',
        'capacity',
        'current_occupancy',
        'status',
        'submitted_by',
        'device_id',
        'is_verified',
    ];

    protected $casts = [
        'latitude' => 'float',
        'longitude' => 'float',
        'capacity' => 'integer',
        'current_occupancy' => 'integer',
        'is_verified' => 'boolean',
    ];

    /**
     * Get the user who submitted this shelter
     */
    public function submitter(): BelongsTo
    {
        return $this->belongsTo(User::class, 'submitted_by');
    }

    /**
     * Check if shelter has available space
     */
    public function hasSpace(): bool
    {
        if ($this->capacity === null) {
            return $this->status === 'open';
        }
        return $this->current_occupancy < $this->capacity && $this->status === 'open';
    }

    /**
     * Get available space count
     */
    public function getAvailableSpaceAttribute(): ?int
    {
        if ($this->capacity === null) {
            return null;
        }
        return max(0, $this->capacity - $this->current_occupancy);
    }
}
